src/components/Chatbar/components/ChatFolders.tsx

```diff
@@ -267,26 +275,43 @@ export const ChatSection = ({
   );
 };
 
-const folderItems: FolderSectionProps<Conversation>[] = [
-  {
-    name: 'Share With Me',
-    itemFilter: SharedWithMeFilter,
-    displayRootFiles: true,
-    dataQa: 'share-with-me',
-  },
-  {
-    name: 'Pinned Chats',
-    itemFilter: PinnedItemsFilter,
-    showEmptyFolders: true,
-    openByDefault: true,
-    dataQa: 'pinned-chats',
-  },
-];
-
 export function ChatFolders() {
+  const { t } = useTranslation(Translation.SideBar);
+  const isFilterEmpty = useAppSelector(
+    ConversationsSelectors.selectIsEmptySearchFilter,
+  );
+  const commonItemFilter = useAppSelector(
+    ConversationsSelectors.selectItemFilter,
+  );
+
+  const isSharingEnabled = useAppSelector((state) =>
+    SettingsSelectors.isFeatureEnabled(state, Feature.ConversationsSharing),
+  );
+
+  const folderItems: FolderSectionProps<Conversation>[] = useMemo(
+    () =>
+      [
+        {
+          hidden: !isSharingEnabled || !isFilterEmpty,
+          name: t('Shared with me'),
+          itemFilter: SharedWithMeFilter,
+          displayRootFiles: true,
+          dataQa: 'share-with-me',
+        },
+        {
+          name: t('Pinned chats'),
+          itemFilter: commonItemFilter,
+          showEmptyFolders: isFilterEmpty,
+          openByDefault: true,
+          dataQa: 'pinned-chats',
+        },
+      ].filter(({ hidden }) => !hidden),
+    [commonItemFilter, isFilterEmpty, isSharingEnabled, t],
+  );
+
   return (
     <div
-      className="flex w-full flex-col gap-0.5 divide-y divide-gray-200 dark:divide-gray-800"
+      className="flex w-full flex-col gap-0.5 divide-y divide-gray-200 empty:hidden dark:divide-gray-800"
```

Comment:
@mikitabut ,
you can investigate code and found that it's almost impossible to check if all children are null because it's responsibility of each section to display itself